src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1445,6 +1466,32 @@ void MulticopterPositionControl::control_auto(float dt)
 	_limit_vel_xy = (!next_setpoint_valid || (_pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LOITER))
 			&& (sqrtf(dist(0) * dist(0) + dist(1) * dist(1)) <= _target_threshold_xy.get());
 
+
+	if (_sonar_measurament.current_distance < _sonar_measurament.max_distance) {
+		if (!_vehicle_land_detected.landed && !in_auto_takeoff()) {
+			if (_obstacle_ahead_sp == false) {
+				stop_sp = _pos;
+				last_obstacle_dist = _sonar_measurament.current_distance;
+				_obstacle_ahead_sp = true;
+			}
+
+
+			if (_obstacle_ahead_sp == true && _sonar_measurament.current_distance < 0.6f) {
```

Comment:
`_obstacle_ahead_sp` is always true here, no? What exactly are you trying to check with this variable? It never gets reset to `false`, so it just checks if `stop_sp` and `last_obstacle_dist` have ever been written.